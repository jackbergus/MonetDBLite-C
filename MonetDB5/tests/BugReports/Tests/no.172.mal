# Monet Mil to Mal compiler
# Copyright (c) 2001-2004, CWI. All rights reserved.

#Predefined code segment
#The compiler can not guarantee an accurate compilation,
#because MIL unlike MAL is a dynamically typed language.
#A few guidelines to maximize usefullness.
#- make the type of variables explicit, in particular 'nil'
#- upon advice of M2m to remap identifiers, change it in your program directly
#use the "mil_schema" for additional support routines
#	io.print("# this used to hang because the no longer persistent\n");
#	io.print("# bat got the BBPUNLOADING status but can only be done\n");
#	io.print("# with refs()==0 and lrefs()==0 \n");
#	io.print("");
	abat := bbp.new(nil:int,nil:int);
#	io.print("var abat := bat(int,int);");
	aa_a := persists(abat,true);
#	io.print("abat.persists(true);");
	ba_a := commit();
#	io.print("# met of zonder 'abat := nil;'\n");
#	io.print("");
	ca_a := getName(abat);
	t_3 := bbp.new(ca_a);
#	io.print("var t_3 := bat(bbpname(abat));");
	da_a := persists(t_3,false);
#	io.print("persists(t_3,false);");
	ea_a := commit();
#	io.print("## Hang:\n");
	fa_a := bat.print(t_3);
#	io.print("");
	ga_a := clients.quit();
#Identifer 'bbpname' mapped to 'getName'
