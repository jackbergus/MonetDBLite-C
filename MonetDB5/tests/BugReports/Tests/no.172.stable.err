stderr of test 'no.172` in directory 'monetdb5/tests/BugReports` itself:


# 21:21:34 >  
# 21:21:34 >  Mtimeout -timeout 60 Mserver --config=/ufs/niels/data/current/trec/MonetDB/Linux/etc/monet.conf --debug=10   --set monet_prompt= --dbname=Tests_tests_BugReports  < no.172.milS
# 21:21:34 >  


# 21:21:35 >  
# 21:21:35 >  Done.
# 21:21:35 >  

