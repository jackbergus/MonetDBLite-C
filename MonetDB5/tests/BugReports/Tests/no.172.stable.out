stdout of test 'no.172` in directory 'monetdb5/tests/BugReports` itself:


# 21:21:34 >  
# 21:21:34 >  Mtimeout -timeout 60 Mserver --config=/ufs/niels/data/current/trec/MonetDB/Linux/etc/monet.conf --debug=10   --set monet_prompt= --dbname=Tests_tests_BugReports  < no.172.milS
# 21:21:34 >  

# Monet Database Server V4.3.5
# Copyright (c) 1993-2002, CWI. All rights reserved.
# compiled for i686-pc-linux-gnu/32bit; dynamically linked.

# this used to hang because the no longer persistent
# bat got the BBPUNLOADING status but can only be done
# with refs()==0 and lrefs()==0 

var abat := bat(int,int).persists(true);
commit();

# met of zonder 'abat := nil;'

var t_3 := bat(bbpname(abat));
persists(t_3,false);
commit();

## Hang:
print(t_3);
#-----------------#
# BAT:	  tmp_26  #
# (int)	  (int)	  #
#-----------------#


# 17:07:33 >  
# 17:07:33 >  Done.
# 17:07:33 >  

